python/MyriaPythonWorker.py

```diff
@@ -123,6 +124,9 @@ def write_with_length(cls, obj, stream, output_type):
         elif output_type == DataType.BLOB:
             cls.write_int(DataType.BLOB, stream)
             cls.pickle_and_write(obj, stream)
+        elif output_type ==DataType.EXCEPTION:
+            cls.write_int(len(obj), stream)
```

Comment:
I don't think this is quite correct in general. The only caller converts a `unicode` object (return value of `format_exc()`) to a UTF-8 `str`, and then passes it to this method, so `len()` works fine. But in general, if this method accepts an arbitrary Python object (i.e., anything that `str()` can format), there's no reason to think that `len(str(obj)) == len(obj)`. So if you want to be able to marshal any object with `DataType.EXCEPTION`, and as long we're stuck with Python 2, I think it would be more consistent for this method to call `unicode(obj).encode("utf-8")` and write the resulting bytes to `stream`. Then the sole caller should not bother encoding the result of `format_exc()`, so should just call `pickle_serializer.write_with_length(traceback.format_exc(), out_file, DataType.EXCEPTION)`.